bodo/libs/streaming/groupby.py

```diff
@@ -164,6 +164,19 @@ def ftypes(self):
             func_types.append(supported_agg_funcs.index(fname))
         return func_types
 
+    def is_precise(self):
+        return self.build_table_type != types.unknown
+
+    def unify(self, typingctx, other):
```

Comment:
Is it possible to move this functionality into the parent class. Something like defining `is_precise` as an abstract method (actually I assume its already defined by the type interface) and just making the common implementation:

```python
if type(self) is type(other):
    if not other.is_precise() and self.is_precise():
        return self
    return other
```

Reply:
I feel like it's safer to have it explicitly here and a few lines of code duplication is worth it for this critical codes.